webvh/webvh/did/manager.py

```diff
@@ -0,0 +1,635 @@
+
+            metadata = resolved_did_doc["metadata"]
+            metadata["state"] = "posted"
+            await event_bus.notify(
+                self.profile,
+                Event(
+                    f"endorsement_response::{endorsed_document['id']}",
+                    {"document": resolved_did_doc["did_document"], "metadata": metadata},
+                ),
+            )
+
+        return resolved_did_doc
+
+    async def update(self, options: dict, features: dict):
+        """Update a Webvh DID."""
+        server_url = self._get_server_info()
+
+        namespace = options.get("namespace")
+        identifier = options.get("identifier")
+
+        if not self._all_are_not_none(namespace, identifier):
+            raise DidCreationError("Namespace and identifier are required.")
+
+        # Get the document state from the server
+        document_state = None
+        try:
+            async for line in self.fetch_jsonl(
+                f"{server_url}/{namespace}/{identifier}/did.jsonl"
+            ):
+                document_state = DocumentState.load_history_line(line, document_state)
+        except ClientResponseError:
+            raise DidCreationError("Failed to fetch the jsonl file from the server.")
+
+        document = document_state.document_copy()
+
+        for feature, values in features.items():
+            document[feature].extend([values])
+
+        author_key_info = await self._get_or_create_author_key(
+            document_state.document["id"]
+        )
+
+        document_state = document_state.create_next(
+            document, {}, datetime.now(timezone.utc).replace(microsecond=0)
+        )
+
+        async with ClientSession() as http_session, self.profile.session() as session:
+            # Submit the log entry
+            log_entry = document_state.history_line()
+            signed_log_entry = await DataIntegrityManager(session).add_proof(
+                log_entry,
+                DataIntegrityProofOptions(
+                    type="DataIntegrityProof",
+                    cryptosuite="eddsa-jcs-2022",
+                    proof_purpose="assertionMethod",
+                    verification_method=f"did:key:{author_key_info.get('multikey')}#{author_key_info.get('multikey')}",
+                ),
+            )
+
+            response = await http_session.post(
+                f"{server_url}/{namespace}/{identifier}",
+                json={"logEntry": signed_log_entry},
+                ssl=False,
+            )
+
+            response_json = await response.json()
+
+            if response_json.get("detail") == "Key unauthorized.":
+                raise DidCreationError("Problem creating log entry: Key unauthorized.")
+
+            resolver = session.inject(DIDResolver)
+
+        return (
+            await resolver.resolve_with_metadata(
+                self.profile, document_state.document["id"]
+            )
+        ).serialize()
+
+    async def deactivate(self, options: dict):
```

Comment:
currently not implemented on the didwebvh server. The route exists, but will return a 501